src/domain/datasets/domain/src/entities/versioned_file.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use file_utils::{MediaType, MediaTypeRef};
+use odf::Multihash;
+
+pub type FileVersion = u32;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[nutype::nutype(derive(AsRef, Clone, Debug, Into))]
+pub struct ExtraDataFields(serde_json::Value);
+
```

Comment:
What did you achieve with this nutype? I don't see any validation or transformation, and the semantic meaning "extra data" is not telling much compared to `some_stuff: serde_json::Value`